realm/realm-library/src/androidTestObjectServer/kotlin/io/realm/RealmFunctionsTests.kt

```diff
@@ -0,0 +1,150 @@
+/*
+ * Copyright 2020 Realm Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package io.realm
+
+import androidx.test.platform.app.InstrumentationRegistry
+import org.bson.*
+import org.bson.codecs.StringCodec
+import org.bson.codecs.configuration.CodecRegistries
+import org.bson.codecs.pojo.PojoCodecProvider
+import org.bson.types.Decimal128
+import org.bson.types.ObjectId
+import org.junit.After
+import org.junit.Before
+import org.junit.Test
+import kotlin.test.assertEquals
+
+class RealmFunctionsTests {
+
+    private lateinit var app: TestRealmApp
+    private lateinit var functions : RealmFunctions
+
+    @Before
+    fun setup() {
+        Realm.init(InstrumentationRegistry.getInstrumentation().targetContext)
+        app = TestRealmApp()
+        functions = RealmFunctions(app.configuration.codecRegistry)
+    }
+
+    @After
+    fun teardown() {
+        if (this::app.isInitialized) {
+            app.close()
+        }
+    }
+
+    // Test of BSON JNI round trip until superseded with actual public api tests are added.
+    @Test
+    fun jniRoundTripForDefaultCodecRegistry() {
+        val i32 = 42
+        val i64 = 42L
+
+        for (type in BsonType.values()) {
+            when (type) {
+                BsonType.DOUBLE -> {
+                    assertTypedEcho(java.lang.Float(1.4), java.lang.Float::class.java)
```

Comment:
What about the primitive types? This is the boxed types if I remember correctly? Same for the other tests

Reply:
Yes. There were some constraints around typing for the `asertTypedEcho` call to match the result. I have updated it to direct `invoke`-calls, to be able to keep the primitive types.